File: native-android-player/app/src/main/java/com/antigravity/player/receiver/BootReceiver.java
package com.antigravity.player.receiver;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.util.Log;
import com.antigravity.player.MainActivity;

/**
 * BootReceiver (Java) - O "dedo no interruptor" do sistema.
 * Migrado para Java para garantir máxima estabilidade em kernels antigos de TV Boxes.
 */
public class BootReceiver extends BroadcastReceiver {
    @Override
    public void onReceive(Context context, Intent intent) {
        String action = intent.getAction();
        
        if (Intent.ACTION_BOOT_COMPLETED.equals(action) ||
            "android.intent.action.QUICKBOOT_POWERON".equals(action) ||
            "com.htc.intent.action.QUICKBOOT_POWERON".equals(action) ||
            "android.intent.action.REBOOT".equals(action)) {
            
            Log.i("SobreMidiaBoot", "Energia restabelecida. Gatilho detectado: " + action);

            // Atraso de Segurança para hardware lento (Opcional, mas recomendado para TV Boxes)
            try {
                Thread.sleep(2000);
            } catch (InterruptedException e) {
                e.printStackTrace();
            }

            // Inicia a MainActivity do Sobre Mídia Player
            Intent i = new Intent(context, MainActivity.class);
            
            // Flags cruciais para abrir o app a partir de background
            i.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
            i.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP);
            i.addFlags(Intent.FLAG_ACTIVITY_SINGLE_TOP);

            context.startActivity(i);
        }
    }
}

File: native-android-player/app/src/main/java/com/antigravity/player/receiver/BootReceiver.kt
package com.antigravity.player.receiver

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.os.Build
import com.antigravity.core.util.Logger
import com.antigravity.player.ui.SplashActivity

class BootReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent) {
        val action = intent.action
        Logger.i("BOOT_RECEIVER", "Intent recebido: $action")

        if (action == Intent.ACTION_BOOT_COMPLETED ||
            action == "android.intent.action.QUICKBOOT_POWERON" ||
            action == "com.htc.intent.action.QUICKBOOT_POWERON"
        ) {
            Logger.i("BOOT_RECEIVER", "Iniciando Sobre Midia Player automaticamente...")
            
            // [O GUARDIÃO DO BOOT]
            // Inicia a SplashActivity para carregar o ciclo natural (Auth -> Main)
            val launchIntent = Intent(context, SplashActivity::class.java).apply {
                addFlags(Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_CLEAR_TASK)
            }
            
            try {
                context.startActivity(launchIntent)
            } catch (e: Exception) {
                Logger.e("BOOT_RECEIVER", "Falha ao iniciar Activity pós-boot: ${e.message}")
            }
        }
    }
}
